app/src/main/java/by/paranoidandroid/notesapp/activities/MainActivity.java

```diff
@@ -28,27 +30,29 @@ protected void onCreate(Bundle savedInstanceState) {
         recyclerView.setLayoutManager(new LinearLayoutManager(this));
         recyclerView.addItemDecoration(new DividerItemDecoration(this, DP_MARGIN));
 
-        adapter = new NotesAdapter(this);
+        adapter = new NotesAdapter(this, this);
         recyclerView.setAdapter(adapter);
-        adapter.updateList(getNoteList());
 
         etTitle =  findViewById(R.id.et_title);
         etBody =  findViewById(R.id.et_body);
 
         Button button = findViewById(R.id.bt_add_note);
         button.setOnClickListener(view -> onAddNoteButtonClick());
+
+        NoteViewModel.Factory viewModelFactory = new NoteViewModel.Factory(getApplication());
```

Comment:
Why did you create a factory?

Reply:
Yes, thos factory is completely useless, removed in https://github.com/Kalyanova/NotesApp/pull/3.